final_prep/mod_zero_hero.rb

```diff
@@ -55,30 +90,75 @@
 #   - citiesDestroyed (array)
 #   - luckyNumbers (array)
 #   - address (hash with following key/values: number , street , state, zip)
+scary_monster = {
+  name: "Razor Burn",
+  smell: "Metallic",
+  weight: 2,
+  cities_destroyed: ["London", "Paris", "New York", "Tokyo"],
+  lucky_numbers: [10, 2, 24, 98],
+  address: {
+    number: 123,
+    street: "Main",
+    state: "Kansas",
+    zip: 67226
+  }
+}
 
 
 # Create a new class called SuperHero
 # - Your class should have the following DYNAMIC values
-#   - name 
+#   - name
 #   - super_power
-#   - age 
+#   - age
 # - Your class should have the following STATIC values
 #   - arch_nemesis, assigned to "The Syntax Error"
 #   - power_level = 100
-#   - energy_level = 50 
+#   - energy_level = 50
+
+class SuperHero
+  attr_accessor :name, :energy_level, :power_level
+  def initialize(name, super_power, age)
+    @name = name
+    @super_power = super_power
+    @age = age
+    @arch_nemesis = "The Syntax Error"
+    @power_level = 100
+    @energy_level = 50
+  end
+
+  # - Create the following class methods
+  #   - say_name, should print the hero's name to the terminal
+  #   - maximize_energy, should update the energy_level to 1000
+  #   - gain_power, should take an argument of a number and INCREASE the power_level by that number
+  def say_name
+    puts "#{name}"
+  end
+
+  def maximize_energy
+    self.energy_level = 1000
+  end
+
+  def gain_power(number_to_increase)
+    self.power_level += number_to_increase
+  end
+end
+
+
+
 
-# - Create the following class methods
-#   - say_name, should print the hero's name to the terminal
-#   - maximize_energy, should update the energy_level to 1000
-#   - gain_power, should take an argument of a number and INCREASE the power_level by that number
 
 # - Create 2 instances of your SuperHero class
+the_typist = SuperHero.new("The Typist", "Typing 180 WPM", 84)
+matrix = SuperHero.new("Matrix", "Ability to slow things down", 25)
 
 
 # Reflection
 # What parts were most difficult about this exerise?
+# It was difficult to remember the syntax and layout of functions and classes.
 
 # What parts felt most comfortable to you?
+# I felt very comfortable declaring variables.
 
 # What skills do you need to continue to practice before starting Mod 1?
-
+# I need to continue to dive into classes and to determine with to use the "@"
+# symbol. I also need to make prototypes for classes and functions.
```

Comment:
Can you expand on what you mean by "make prototypes for classes and functions"?

When I hear this, I think of JavaScript. In ruby we don't need prototypes because it is an OOP language :)